flow/record/fieldtypes/__init__.py

```diff
@@ -50,9 +55,15 @@ def flow_record_tz(*, default_tz: str = "UTC") -> Optional[ZoneInfo | UTC]:
     Returns:
         None if ``FLOW_RECORD_TZ=NONE`` otherwise ``ZoneInfo(FLOW_RECORD_TZ)`` or ``UTC`` if ZoneInfo is not found.
     """
+
     tz = os.environ.get("FLOW_RECORD_TZ", default_tz)
     if tz.upper() == "NONE":
         return None
+
+    if not HAS_ZONE_INFO:
+        warnings.warn("Cannot use FLOW_RECORD_TZ due to missing zoneinfo module, defaulting to 'UTC'.")
```

Comment:
No warning needs to be given in these cases:

* FLOW_RECORD_TZ is not set, it will default to "UTC"
* FLOW_RECORD_TZ is set to "UTC"

In those cases it already returns UTC, so no warnings needs to be given. This ensures that in default behaviour no unnecessary warnings are triggered.

```suggestion
        if tz != "UTC":
            warnings.warn("Cannot use FLOW_RECORD_TZ due to missing zoneinfo module, defaulting to 'UTC'.")
```
Note `tz` here is a string from the environment variable `FLOW_RECORD_TZ` and not a timezone object